packages/models/src/appMapBuilder/index.js

```diff
@@ -190,37 +195,45 @@ class AppMapBuilder extends EventSource {
       // Build a list of unique exclusions, starting with the largest event
       // type. Iterate until the estimated event array size is under our
       // threshold.
-      const exclusions = new Set();
       for (let i = 0; i < eventAggregates.length; i += 1) {
         const eventInfo = eventAggregates[i];
         if (eventInfo.totalBytes <= totalBytes * pruneRatio) {
           break;
         }
-        exclusions.add(eventInfo.id);
+        this.data.exclusions.add(eventInfo.id);
       }
+    }).chunk((stacks) => this.excludeEvents(stacks, classMap, this.data.exclusions));
+  }
 
-      return stacks.map((events) =>
-        events.filter((e) => {
-          const { callEvent } = e;
-
-          // If there's no call event, there's no need to retain this event
-          if (!callEvent) {
-            return false;
-          }
-
-          if (
-            callEvent.http_server_request ||
-            callEvent.http_client_request ||
-            callEvent.sql_query
-          ) {
-            return true;
-          }
-
-          const name = eventName(classMap, e);
-          return !exclusions.has(name);
-        })
-      );
-    });
+  removeEvents(events) {
+    let exclusions;
+    let classMap;
+
+    return this.on('preprocess', (d) => {
+      exclusions = new Set(events);
+      classMap = new ClassMap(d.data.classMap);
+    }).chunk((stacks) => this.excludeEvents(stacks, classMap, exclusions));
+  }
+
+  // expects exclusions to be a Set
+  excludeEvents(stacks, classMap, exclusions) {
+    return stacks.map((events) =>
+      events.filter((e) => {
+        const { callEvent } = e;
+
+        // If there's no call event, there's no need to retain this event
+        if (!callEvent) {
+          return false;
+        }
+
+        if (callEvent.http_server_request || callEvent.http_client_request || callEvent.sql_query) {
+          return true;
+        }
+
+        const name = eventName(classMap, e);
```

Comment:
`eventName` is

````${callEvent.defined_class}${callEvent.static ? '.' : '#'}${callEvent.method_id}````

Nowadays, we don't use this data any more for filter. A better option is to use `CodeObject#fqid`. I think now is a good time to make that change. 

In fact, probably a better alternative is to allow the command to accept an `appMapFilter` as a command line (e.g. serialized to JSON or Base64) and simply apply that rather than re-implementing a different kind of filtering.